### ISSUE TITLE
Attachment download progress indicator with ability to cancel

### ISSUE THREAD

**Issue by aurimasv** [zotero/zotero#641]
Re https://forums.zotero.org/discussion/46864

The goal is to display download progress, total PDF size (where available), and ability to cancel when importing items from web (and search?) translators.

One option is to do this in the current progress indicator (which needs some tweaking too, but that's another issue). I would put a short row underneath each attachment. In that row, we would have a linear progress bar (as wide as can fit), total size, and a red X to cancel the download. Once the download is complete, we can remove the progress row. If possible, we could display a mouse-over popup for download ETA, though I don't think that's too critical.

This does take up some additional vertical real estate, which is sometimes sparse, but I think we need general improvements in this regard to accommodate for large imports (I think we need to make the progress popup scrollable)

**Comment by dstillman** [comment]
By the "current progress indicator", you mean the translator save popup in the bottom-right? The main problem with that is that it's easily dismissible — I think just clicking on it makes it go away. We could make it persist until everything is finished, but it should possibly then be moveable in case it's in the way.

**Comment by aurimasv** [comment]
Yes, that's what I mean. The easily dismissible part is something I'm not a big fan of and I agree that that would have to be adjusted as well.

**Comment by aurimasv** [comment]
But I suppose to go back to what you were also suggesting on the forums, we need some sort of way to manage these downloads from within Zotero client as well. I think a general mechanism to manage and display background tasks is needed.

**Comment by simonster** [comment]
We have an attachment progress indicator when the connectors save to the server that replaces the attachment icon with a circle that progressively fills in as the translator saves. We make the progress indicator turn into an X on mouseover and cancel if it's clicked, and we could show the file size in a tooltip.

**Comment by simonster** [comment]
Screenshot of what this looks like: ![](http://i.imgur.com/so1HVb3.png)

**Comment by dstillman** [comment]
We also have the little pie that we use for sync downloads.
